koswat/cost_report/io/summary/summary_infrastructure_costs/summary_infrastructure_costs_csv_fom_builder.py

```diff
@@ -0,0 +1,201 @@
+from collections import defaultdict
+from typing import Any
+
+from koswat.core.io.csv.koswat_csv_multi_header_fom import KoswatCsvMultiHeaderFom
+from koswat.core.protocols.builder_protocol import BuilderProtocol
+from koswat.cost_report.infrastructure.infrastructure_location_costs import (
+    InfrastructureLocationCosts,
+)
+from koswat.cost_report.infrastructure.infrastructure_location_profile_cost_report import (
+    InfrastructureLocationProfileCostReport,
+)
+from koswat.cost_report.multi_location_profile.multi_location_profile_cost_report import (
+    MultiLocationProfileCostReport,
+)
+from koswat.cost_report.summary.koswat_summary import KoswatSummary
+from koswat.dike.surroundings.point.point_surroundings import PointSurroundings
+from koswat.strategies.strategy_location_reinforcement import (
+    StrategyLocationReinforcement,
+)
+
+
+class SummaryInfrastructureCostsCsvFomBuilder(BuilderProtocol):
+    koswat_summary: KoswatSummary
+    _ordered_profile_types: list[str]
+    _ordered_infra_types: list[str]
+
+    def __init__(self) -> None:
+        self.koswat_summary = None
+        self._ordered_profile_types = []
+        self._ordered_infra_types = []
+
+    def build(self) -> KoswatCsvMultiHeaderFom:
+        self._ordered_profile_types = list(
+            map(
+                lambda x: x.profile_type_name,
+                self.koswat_summary.locations_profile_report_list,
+            )
+        )
+        self._ordered_infra_types = sorted(
+            list(
+                set(
+                    _ilpcr.infrastructure.infrastructure_name
+                    for _ilpcr in self.koswat_summary.locations_profile_report_list[
+                        0
+                    ].infra_multilocation_profile_cost_report
+                )
+            )
+        )
+
+        _csv_fom = KoswatCsvMultiHeaderFom()
+        _csv_fom.headers = self._get_headers()
+        _csv_fom.entries = self._get_locations_matrix(
+            self.koswat_summary.reinforcement_per_locations,
+            self.koswat_summary.locations_profile_report_list,
+        )
+
+        return _csv_fom
+
+    def _get_locations_matrix(
+        self,
+        reinforcement_per_locations: list[StrategyLocationReinforcement],
+        locations_profile_report_list: list[MultiLocationProfileCostReport],
+    ) -> list[list[Any]]:
+        def _get_totals(location: PointSurroundings) -> list[float]:
+            _totals = []
+            for _profile_type in self._ordered_profile_types:
+                _totals.append(
+                    sum(
+                        x.total_cost
+                        for x in _infra_cost_per_loc_dict[location][
+                            _profile_type
+                        ].values()
+                    )
+                )
+            return _totals
```

Comment:
so ?

```suggestion
            return list(
                    sum(
                        x.total_cost
                        for x in _infra_cost_per_loc_dict[location][
                            _profile_type
                        ].values()
                    )
                    for _profile_type in self._ordered_profile_types
                )
```


or ...

```suggestion
            def total_cost_per_profile_and_location(profile_type) -> float:
                return sum(map(lambda x: x.total_cost, _infra_cost_per_loc_dict[location][profile_type].values()))

            return list(map(total_cost_per_profile_and_location, self._ordered_profile_types))
```

Reply:
Thanks, like the first more.